studio/pages/join/[slug].tsx

```diff
@@ -0,0 +1,153 @@
+import React from 'react'
+import { observer } from 'mobx-react-lite'
+import { useStore, withAuth } from 'hooks'
+import { Button, Typography } from '@supabase/ui'
+import { useState } from 'react'
+import { useRouter } from 'next/router'
+import Link from 'next/link'
+import { API_URL } from 'lib/constants'
+import { get, post, delete_ } from 'lib/common/fetch'
+import { useEffect } from 'react'
+
+interface TokenInfo {
+  organization_name: string | undefined
+  token_does_not_exist: boolean
+  email_match: boolean
+  authorized_user: boolean
+  expired_token: boolean
+  invite_id: number
+}
+
+const User = () => {
+  const router = useRouter()
+  const { slug, token } = router.query
+  const { ui } = useStore()
+  const [isSubmitting, setIsSubmitting] = useState(false)
+  const [tokenValidationInfo, setTokenValidationInfo] = useState<TokenInfo>()
+  const {
+    token_does_not_exist,
+    email_match,
+    authorized_user,
+    expired_token,
+    organization_name,
+    invite_id,
+  } = tokenValidationInfo || {}
+
+  useEffect(() => {
+    let cancel = false
+
+    async function fetchTokenInfo() {
+      const response = await get(`${API_URL}/organizations/${slug}/members/join?token=${token}`, {})
+      if (!cancel) setTokenValidationInfo(response)
+    }
+
+    if (router.query.token) {
+      fetchTokenInfo()
+    }
+
+    return () => {
+      cancel = true
+    }
+  }, [])
+
+  async function handleJoinOrganization() {
+    setIsSubmitting(true)
+    const response = await post(`${API_URL}/organizations/${slug}/members/join?token=${token}`, {})
+    if (response.error) {
+      ui.setNotification({
+        category: 'error',
+        message: `Failed to join organization: ${response.error.message}`,
+      })
+      setIsSubmitting(false)
+    } else {
+      setIsSubmitting(false)
+      router.push('/')
```

Comment:
@joshenlim after you accept the invite, you get brought to `/`. But the list of orgs in the sidebar doesn't change until you refresh. Any idea what we can do about that? 

First load:
<img width="275" alt="CleanShot 2022-06-17 at 14 29 18@2x" src="https://user-images.githubusercontent.com/105593/174344134-ad0ce781-4fdf-4d31-bfdb-2f43b242e6fb.png">

Subsequent loads:
<img width="259" alt="CleanShot 2022-06-17 at 14 30 35@2x" src="https://user-images.githubusercontent.com/105593/174344178-de78d078-dbbc-4770-a72a-34507065b234.png">

Reply:
Missed this comment earlier! I get what you mean now, thanks! I’ll check with api team tomorrow.

Reply:
this is possibly all mute now, because the join page doesn't load any stores now, so when it redirects the app will be forced to load everything anyway.

Reply:
No, sadly still need something here.  It's working now b/c I'm using `app.organizations.load()` when you join an org. That refreshes the store and pulls in the updated list of orgs.